android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -954,6 +954,62 @@ class GeckoEngineSession(
         })
     }
 
+    /**
+     * See [EngineSession.getNeverTranslateSiteSetting]
+     */
+    override fun getNeverTranslateSiteSetting(
+        onResult: (Boolean) -> Unit,
+        onException: (Throwable) -> Unit,
+    ) {
+        if (geckoSession.sessionTranslation == null) {
+            onException(sessionTranslationNotAvailable())
+            return
+        }
+
+        geckoSession.sessionTranslation!!.neverTranslateSiteSetting.then({
+                response ->
+            val errorMessage = "Did not receive a site setting response."
```

Comment:
Nit: Maybe we should move `errorMessage` to the `if` statement and it's only used there.

Reply:
Updated, thanks!